### DISCUSSION TITLE
Feature request: repository_dispatch to any specified branch

### DISCUSSION THREAD

**Discussion by ReenigneArcher** [community/community#50436] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

Currently repository dispatch will not trigger an event unless the file exists on the default branch. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#repository_dispatch

This is a fairly substantial limitation for projects that use a staging branch before publishing changes to the default branch.

I would like to request the ability to specify the target branch of the repository dispatch event.

A similar discussion here, but unresolved: https://github.com/orgs/community/discussions/24657

**Comment by huehnerlady** [comment]
It also makes creating new github actions tricky as you cannot try them out before you go through the review process
